### ISSUE TITLE
bibtex file as bundle module -- installed in wrong place

### ISSUE THREAD

**Issue by leingang** [latex3/l3build#295]
I have an l3build bundle with a module containing a bibtex file. This isn't a bibtex file for typesetting documentation (although I may want to make such documentation); it's a bibtex file to be installed in the user's texmf tree for including in other documents.

    foo
    |- build.lua
    |- bibtex
        |- bar.bib
        |- build.lua

The `build.lua` file looks like this:

```lua
bundle = "foo"
module = "bibtex"
maindir = ".."
```

I expected that `l3build install` would send `bar.bib` to `${TEXMFHOME}/bibtex/bib/foo/bar.bib`. Instead, it does nothing. `l3build install --full` sends it to `doc/latex/foo/bibtex/bib/bar.bib`. This means `bibtex` can't find it. Is my configuration wrong?

I'm using l3build version 2023-02-20 on TeXLive 2022.

**Comment by leingang** [comment]
> bib files are assumed doc

That makes sense, but it seems contrary to the stated behavior in Section 4.1 of the manual:

> With the standard settings, l3build will install files within the TEX directory structure (TDS) as follows:
> * `installfiles` within a ⟨bundle⟩/⟨module⟩ (or⟨module⟩) directory inside `tex/⟨format⟩`
> * `sourcefiles` within a ⟨bundle⟩/⟨module⟩(or ⟨module⟩) directory inside `source/⟨format⟩`
> * Typeset PDFs within a⟨bundle⟩/⟨module⟩(or⟨module⟩) directory inside `doc/⟨format⟩`
> * `bstfiles` within a ⟨bundle⟩/⟨module⟩ (or ⟨module⟩) directory inside `bibtex/bst` 
> * `bibfiles` within a ⟨bundle⟩/⟨module⟩ (or ⟨module⟩) directory inside `bibtex/bib` [not `doc/<format>/bibtex/bib`]
> * `makeindexfiles` within a ⟨bundle⟩/⟨module⟩ (or ⟨module⟩) directory inside `makeindex`

Setting `tdslocations` works around well:

```lua
tdslocations = {"bibtex/bib/" .. bundle .. "/*.bib"}
```

**Comment by FrankMittelbach** [comment]
> bib files are assumed doc, you can add all .bib or just that one to installfiles and specify a tds location see https://github.com/ho-tex/oberdiek/blob/main/build.lua (which may not be a model of best practice, but does work)

they are part of the documentation but part of the documentation source, so I think they should end up under source because 99.9% of the time they will be project specifict and not of general interest. They should therefore not end up in the searched texmf tree as this would generate chaos most of the time. But I don't think they should end up below /doc but below /source next to the sources that use them as part of the documentation. So in my opinion both the l3build documentation as well as what it does in reality do is slightly wrong.

**Comment by josephwright** [comment]
@FrankMittelbach  My understanding is that `source` is meant to be 'source for the 'live' part of the tree', while `doc` is for 'documentation and sources only needed by documentation'. That's where for example I've always put the demo `.bib` files for `achemso`, `biblatex-<thing>`, etc.

**Comment by josephwright** [comment]
@FrankMittelbach  The tricky part is we _do_ put `.tex` files into `doc` where they are purely needed to generate the docs. For a `.bib` file, it _could_ go with a `.dtx` or it _could_ go with a `.tex` that lives inside `doc`: there's not a clear ine. So I don't think there's a definite clear choice here: perhaps best to simply document the current behaviour and say that users may need to adjust?

**Comment by FrankMittelbach** [comment]
yes I saw that we _do_ that. But is that because we deliberately do this or because somebody coded the base distribution this way because the assumption was it should be this way? I'm not saying you are wrong, I think we both agree that the bib file should go next to the doc sources, the question is where by default should go the doc sources? As I can see in other repos it doesn't seem to be under doc, but maybe I just looked at the wrong examples and the situation is not as clear cut as I think. Anyhow, right now the bib files go by default to doc but the doc sources, eg the dtx go by default to source, don't they?
